pandas/core/arrays/interval.py

```diff
@@ -588,7 +595,7 @@ def __eq__(self, other):
         if is_interval_dtype(other_dtype):
             if self.closed != other.closed:
                 return np.zeros(len(self), dtype=bool)
-            return (self.left == other.left) & (self.right == other.right)
+            return (self._left == other.left) & (self._right == other.right)
```

Comment:
Is `other` here known to be a particular type (like `IntervalArray`), or is it something like `Union[IntervalArray, Series, Index,Interval]`. If it's just IntervalArray it'd be a bit faster to compare with `other._left` and `right._left`.

Reply:
at this point `other` can be an Interval or IntervalArray